core/src/main/java/com/sequenceiq/cloudbreak/repository/HostGroupRepository.java

```diff
@@ -36,11 +36,11 @@
     @Query("SELECT h FROM HostGroup h JOIN h.recipes r WHERE r.id= :recipeId")
     Set<HostGroup> findAllHostGroupsByRecipe(@Param("recipeId") Long recipeId);
 
-    @Query("SELECT h FROM HostGroup h LEFT JOIN FETCH h.recipes WHERE h.cluster.id= :clusterId")
+    @Query("SELECT h FROM HostGroup h LEFT JOIN FETCH h.recipes LEFT JOIN FETCH h.generatedRecipes WHERE h.cluster.id= :clusterId")
```

Comment:
yes it's required as this one is accessed from the hostGroup not from the recipe, so that's needed in order to avoid lazy init exceptions. the for recipe <-> generated recipe relation, it's more like informative to have the entities on both side (similar as with hostGroup <-> generated recipe, so orphan removal can be defined on recipe side)